idmapping-rest/src/main/java/org/uniprot/api/idmapping/repository/IdMappingRepository.java

```diff
@@ -0,0 +1,88 @@
+package org.uniprot.api.idmapping.repository;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.validation.constraints.Size;
+
+import lombok.AllArgsConstructor;
+
+import org.apache.solr.client.solrj.SolrClient;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.common.params.MapSolrParams;
+import org.springframework.stereotype.Repository;
+import org.uniprot.api.idmapping.model.IdMappingStringPair;
+import org.uniprot.store.search.SolrCollection;
+
+@Repository
+@AllArgsConstructor
+public class IdMappingRepository {
+
+    private static final int MAX_ID_MAPPINGS_ALLOWED = 100_000;
+    private final SolrClient uniProtKBSolrClient;
+    private final SolrClient solrClient;
+
+    public List<IdMappingStringPair> getAllMappingIds(
+            SolrCollection collection, @Size(max = MAX_ID_MAPPINGS_ALLOWED) List<String> searchIds)
+            throws SolrServerException, IOException {
+        switch (collection) {
+            case uniprot:
+                return getAllMatchingIds(
+                        uniProtKBSolrClient, collection, "all_ids", "accession_id", searchIds);
+            case uniparc:
+                return getAllMatchingIds(solrClient, collection, "upi", "upi", searchIds);
+            case uniref:
+                // uniref id is big (23 char e-g UniRef100_UPI0000072840) 100_000 can not fit in
+                // single request
+                var sublistSize = Math.min(searchIds.size(), MAX_ID_MAPPINGS_ALLOWED / 2);
```

Comment:
We should avoid switch case it violates SOLID principle. It violates Single responsibility principle and open close principle as well.

Reply:
In general yes, But here no

Reply:
Would you mind explaining why not?
i can see this class has more than one reason to change. - Violation of SRP
Tomorrow if we support more internal mapping like supporting data, we need to change this code. - Violation of OCP.